crates/bevy_reflect/src/type_registry.rs

```diff
@@ -406,6 +406,15 @@ impl TypeRegistry {
     pub fn iter_mut(&mut self) -> impl Iterator<Item = &mut TypeRegistration> {
         self.registrations.values_mut()
     }
+
+    /// Checks to see if the [`TypeData`] of type `T` is associated with each registered type,
+    /// returning a ([`TypeRegistration`], [`TypeData`]) iterator for all entries where data of that type was found.
+    pub fn iter_with_data<T: TypeData>(&self) -> impl Iterator<Item = (&TypeRegistration, &T)> {
+        self.registrations.values().filter_map(|item| {
+            let type_data = self.get_type_data::<T>(item.type_id());
```

Comment:
This incurs an extra lookup (to get the registration). We can avoid this by just using `item` directly:
```suggestion
            let type_data = item.data::<T>(item.type_id());
```